packages/transformer/src/IModelTransformer.ts

```diff
@@ -2890,19 +2914,64 @@ export class IModelTransformer extends IModelExportHandler {
         )
           continue;
         await this.processDeletedOp(
-          change,
+          change.ECInstanceId,
+          change.$meta?.classFullName ?? "",
+          change.FederationGuid,
           elemIdToScopeEsa,
           relationshipECClassIds.has(ecClassId ?? ""),
+          change.SourceECInstanceId,
+          change.TargetECInstanceId,
           alreadyImportedElementInserts,
           alreadyImportedModelInserts
         );
+        deleteIdsProcessed.add(change.ECInstanceId);
       }
 
       csReader.close();
     }
+
+    // Because there is a possibility that someone could manually add ids to exporter.sourceDbChanges.deleteIds, we must separately process exporter.sourceDbChanges.deleteIds as deletedOps.
+    if (this.exporter.sourceDbChanges?.hasCustomChanges) {
+      for (const changedInstances of changedInstanceOps) {
+        for (const id of this.exporter.sourceDbChanges?.[
+          changedInstances
+        ].deleteIds.keys() ?? []) {
+          // If id was already processed, that means it was a normal delete found in a changeset. We can skip it.
+          if (deleteIdsProcessed.has(id)) continue;
+          const customData = this.exporter.sourceDbChanges?.getCustomDataFromId(
+            id,
+            changedInstances
+          );
+          if (customData) {
+            const ecClassId = customData?.ecClassId;
+            const classFullName =
+              this.exporter.sourceDbChanges?.getClassFullNameFromECClassId(
+                ecClassId
+              );
+            const fedGuid = customData?.federationGuid;
+            const sourceIdOfRelationshipInSource =
+              customData?.sourceIdOfRelationship;
+            const targetIdOfRelationshipInSource =
+              customData?.targetIdOfRelationship;
+            await this.processDeletedOp(
```

Comment:
[Q] Why processDeleteOp() is called in processChangesets() for custom and non custom changes? Aren't deleted elements also hanlded in Exporter:exportchanges()?
![image](https://github.com/user-attachments/assets/dba98049-d4ca-4885-87ae-6c1ad1f07a2c)

Reply:
processDeletedOp's purpose is not to actually propagate the deletes like the exporter.exportChanges does. It mostly serves to build up the remap table for these deleted elements. We do attempt to build up the remap table in general at the top of `processChangesets`, but it wont catch any elements that no longer exist (because they were deleted).

I believe it also helps us handle the case where an element is deleted and then inserted again within a range of changesets and prevents us from accidentally deleting it from the target, when it should actually be an update in the target instead.